modules/highlight/highlight.go

```diff
@@ -73,11 +84,18 @@ func Code(fileName, code string) string {
 		lexer = lexers.Get(val)
 	}
 
+	if lexer == nil {
+		if l, ok := cache.Get(fileName); ok {
```

Comment:
Is the fileName an absolute path?

Reply:
I don't think that matters as it only gets lexer

Reply:
To be honest, I am not so sure. 
As far as I can tell, the matcher of `chroma` cares about the file extension only. It uses the extension to determine the file type for syntax highlighting.

Reply:
The cache will be big more and more except Gitea restarted. Maybe we should get only ext of the filename here.